src/transaction/txParser.ts

```diff
@@ -43,13 +47,51 @@ const parseTxInputs = (
   return txInputs.map(([txHash, outputIndex]): _Input => ({ txHash, outputIndex }))
 }
 
-const parseTxOutputs = (
-  txOutputs: any[],
-): _Output[] => {
+const parseAssets = (assets: any): _Asset[] => {
+  if (!(assets instanceof Map)) {
+    throw Error(Errors.TxAssetParseError)
+  }
+  return Array.from(assets).map(([assetName, coins]) => {
+    if (!Buffer.isBuffer(assetName)) {
+      throw Error(Errors.AssetNameParseError)
+    }
+    // TODO: is lovelace is not the best name since its not a lovelace
+    if (!isLovelace(coins)) {
```

Comment:
isUint64 would imho be an ideal generalization that can be applied to ttl/validity start and possibly some other values

Reply:
Refactored